### PR TITLE
En 4444/processor factories and tx coordinator metachain

### PR DESCRIPTION
**Pull request by sasurobert**
Factories and tx Cooridnator creation for metachain

### CODE REVIEW

File: core/address.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package core
+
+import (
+	"bytes"
+	"github.com/ElrondNetwork/elrond-go/process/smartContract/hooks"
```
Comment (bogdan-rosianu) [inline core/address.go]: goimports

File: cmd/node/factory/structs.go [hunk 2]
```diff
@@ -7,7 +7,6 @@ import (
 	"encoding/hex"
```
Comment (bogdan-rosianu) [inline cmd/node/factory/structs.go]: in core/address.go:
-> fix imports
-> lines 59-63 could be replaced with `return isOnMetaChainSCAddress`

File: process/factory/metachain/feeHandler.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import "math/big"
+
+type TransactionFeeHandler struct {
+}
+
+func (t *TransactionFeeHandler) ProcessTransactionFee(cost *big.Int) {
+}
+
+func (t *TransactionFeeHandler) IsInterfaceNil() bool {
```
Comment (bogdan-rosianu) [inline process/factory/metachain/feeHandler.go]: No comments for exported fields

File: process/factory/metachain/preProcessorsContainerFactory_test.go [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+		&mock.TxProcessorMock{},
+		&mock.FeeHandlerStub{},
+	)
+
+	assert.Nil(t, err)
+	assert.NotNil(t, ppcm)
+}
+
+func TestPreProcessorsContainerFactory_CreateErrTxPreproc(t *testing.T) {
+	t.Parallel()
```
Comment (bogdan-rosianu) [inline process/factory/metachain/preProcessorsContainerFactory_test.go]: add empty line

File: process/factory/metachain/intermediateProcessorsContainerFactory_test.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+		&mock.HasherMock{},
+		&mock.AddressConverterMock{},
+		&mock.ChainStorerMock{},
+	)
+
+	assert.Nil(t, err)
+	assert.NotNil(t, ipcf)
+}
+
+func TestIntermediateProcessorsContainerFactory_Create(t *testing.T) {
```
Comment (bogdan-rosianu) [inline process/factory/metachain/intermediateProcessorsContainerFactory_test.go]: maybe you can add a test with a negative scenario (like `TestPreProcessorsContainerFactory_CreateErrTxPreproc`)

File: process/factory/metachain/preProcessorsContainerFactory_test.go [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+		&mock.RequestHandlerMock{},
+		&mock.TxProcessorMock{},
+		nil,
+	)
+
+	assert.Equal(t, process.ErrNilEconomicsFeeHandler, err)
+	assert.Nil(t, ppcm)
+}
+
+func TestNewPreProcessorsContainerFactory_NilTxProcessor(t *testing.T) {
```
Comment (bogdan-rosianu) [inline process/factory/metachain/preProcessorsContainerFactory_test.go]: maybe rearrange the tests in order to check nil parameters in order ? (tip: select a test name and ctrl+shift+up)

File: core/address.go (hunk 1 above)
Comment (sasurobert) [reply core/address.go]: done

File: cmd/node/factory/structs.go (hunk 2 above)
Comment (sasurobert) [reply cmd/node/factory/structs.go]: done

File: process/factory/metachain/feeHandler.go (hunk 3 above)
Comment (sasurobert) [reply process/factory/metachain/feeHandler.go]: done

File: process/factory/metachain/preProcessorsContainerFactory_test.go (hunk 4 above)
Comment (sasurobert) [reply process/factory/metachain/preProcessorsContainerFactory_test.go]: done

File: process/factory/metachain/preProcessorsContainerFactory_test.go (hunk 6 above)
Comment (sasurobert) [reply process/factory/metachain/preProcessorsContainerFactory_test.go]: not critical.

File: process/factory/metachain/intermediateProcessorsContainerFactory_test.go (hunk 5 above)
Comment (sasurobert) [reply process/factory/metachain/intermediateProcessorsContainerFactory_test.go]: already added.

Comment (ccorcoveanu) [reply process/factory/metachain/intermediateProcessorsContainerFactory_test.go]: Is it not pushed yet?
